examples/JsEventHandlerExamples/transformLayout-addField.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Name: Add Field Transform Layout Event Handler
+ * Description: This event handler adds a field to the layout programmatically.
+ * Note: This may need to be paired with a replaceSave event handler to ensure that the field is saved.
+ */
+(function(convenienceApi, EVENT_NAMES, privilegedEnvelope = {}) {
+
+	const eh = {};
+	
+	function addFieldToLayout(layoutData) {
+        try {
+			const FIELD_TO_ADD = {
+				"FieldType": "SingleObject",
+				"DisplayType": "Dropdown",
+				"ArtifactTypeID": 1000056,
+				"PickerViewArtifactID": 0,
+				"FieldID": 1040289, // this is Artifact ID but if the field is not real, it can be a string too
+				"DisplayName": "Test RDO Single Object",
+				"AllowCopyFromPrevious": false,
+				"ShowNameColumn": true,
+				"IsReadOnly": false,
+				"IsRequired": false,
+				"IsSystem": false,
+				"FieldCategoryID": 0,
+				"EnableDataGrid": false,
+				"Guids": [], // you can populate your guid(s) here and use them in handlers
+				// 1-indexed placement info
+				// you'll change this to match where you'd like them to be in the layout
+				// layouts are top-down, left-right (so Row is the most significant bit)
+				"Colspan": 1, // one column or two?
+				"Column": 1,
+				"Row": 1
+			};
+
+			const fields = convenienceApi.layout.getFields(layoutData);
+			const needsToAddField = !fields.find(f => f.DisplayName === FIELD_TO_ADD.DisplayName); 
+
+			if (!needsToAddField) {
+				console.log(`Field ${FIELD_TO_ADD.DisplayName} already exists -- skipping addition of new field.`);
+			} else {
+				// should be useful for adding fields to json
+				const categories = convenienceApi.layout.getCategories(layoutData);
+
+				// when you add a field, are you adding it to the top of a category, the bottom of a category,
```

Comment:
Could this comment be moved to the top of the file to spell out the prerequisites for this event  handler? Looks like we're assuming the layout has a category named "TR Text Two"